bundles/io.openliberty.tools.eclipse.ui/src/io/openliberty/tools/eclipse/debug/DebugModeHandler.java

```diff
@@ -228,6 +232,27 @@ protected IStatus run(IProgressMonitor monitor) {
                         return Status.CANCEL_STATUS;
                     }
 
+                    String debugPort = null;
+
+                    if (readDebugPort) {
+                        // Read debug port from server.env. If devmode has restarted the server, it may
+                        // take a few seconds for it to find a new port in the event that the previous port
+                        // is still being held by the OS.
+                        Thread.sleep(5000);
```

Comment:
Maybe let's try 10 sec.   Even if we attach the debugger early, the app might not be started.  By this logic it might be worth waiting in case dev mode restarting the server hasn't progressed to updating the debug port to a new selection. 

We might want to come back to this and think through what happens if we're wrong.   Do we fail fast  or hang or something else?

Reply:
Thinking through this - if we are too early and read the wrong port (e.g. the old port in server.env before dev mode updated), our debugger will try to connect and timeout after 3 minutes. After that, if a user disconnects the debugger, it will auto-reconnect and will read the new port.  If this becomes a problem, we can look to tweak that 3 minutes and possible fail faster.